src/mr_allocator.rs

```diff
@@ -72,12 +72,24 @@ static mut RDMA_EXTENT_HOOKS: extent_hooks_t = extent_hooks_t {
     merge: Some(RDMA_MERGE_EXTENT_HOOK),
 };
 
+/// Combination of `ProtectionDomain`'s ptr and `ibv_access_falgs` as a key.
+#[derive(PartialEq, Eq, Hash)]
+pub(crate) struct AccessPDKey(u128);
+
+impl AccessPDKey {
+    /// Cast the `ProtectionDomain`'s ptr and `ibv_access_flags` into a `u128`
+    #[allow(clippy::as_conversions)]
+    pub(crate) fn new(pd: &Arc<ProtectionDomain>, access: ibv_access_flags) -> Self {
```

Comment:
Why don't we merge protection domain id and flags? And why doesn't a tuple work here?

Reply:
A tuple means `(protection domain id, flags)`

Reply:
Oh, tuple is certainly workable, why I wrote this ┑(￣。。￣)┍